pkg/api/http/directmessaging_test.go

```diff
@@ -727,6 +728,36 @@ func TestV1DirectMessagingEndpoints(t *testing.T) {
 		assert.Equal(t, "ERR_DIRECT_INVOKE", resp.ErrorBody["errorCode"])
 	})
 
+	t.Run("Invoke returns error - 500 NO TARGET_APP_ID ", func(t *testing.T) {
+		apiPath := "fakeMethod"
+		fakeData := []byte("fakeData")
+
+		mockDirectMessaging.Calls = nil // reset call count
+
+		mockDirectMessaging.
+			On(
+				"Invoke",
+				mock.MatchedBy(matchContextInterface),
+				mock.MatchedBy(func(b string) bool {
+					return b == "fakeAppID"
+				}),
+				mock.AnythingOfType("*v1.InvokeMethodRequest"),
+			).
+			Return(nil, errors.New("NO TARGET_APP_ID")).
+			Once()
```

Comment:
The reason this test causes failures if you move it executes earlier, is because the .On() calls pile up and the later tests catch up the expected calls defined in the test in question. So you could either restart the mock for every test, or make the expected call more constrained.

```suggestion
		apiPath := "noAppIDMethod"
		fakeData := []byte("fakeData")

		mockDirectMessaging.Calls = nil // reset call count

		mockDirectMessaging.
			On(
				"Invoke",
				mock.MatchedBy(matchContextInterface),
				mock.MatchedBy(func(b string) bool {
					return b == "noAppIDMethod"
				}),
				mock.AnythingOfType("*v1.InvokeMethodRequest"),
			).
			Return(nil, errors.New("NO TARGET_APP_ID")).
			Once()
```

Reply:
I see, thanks for explaining this.